From cd83ad4fbde4c3ad4da7b179a5aada257f3c7071 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 23 Feb 2017 08:18:36 +0000
Subject: [PATCH] policy/modules/system/sysnetwork: make dhcpc_t domain MLS
 trusted for reading from files up to its clearance

Allow dhcpc_t to search /run/systemd/journal

Fixes:
avc:  denied  { search } for  pid=218 comm="dhclient" name="journal"
dev="tmpfs" ino=10990 scontext=system_u:system_r:dhcpc_t:s0-s15:c0.c1023
tcontext=system_u:object_r:syslogd_runtime_t:s15:c0.c1023 tclass=dir
permissive=0

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/sysnetwork.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/sysnetwork.te b/policy/modules/system/sysnetwork.te
index a77738924..62c98446d 100644
--- a/policy/modules/system/sysnetwork.te
+++ b/policy/modules/system/sysnetwork.te
@@ -154,6 +154,8 @@ sysnet_run_ifconfig(dhcpc_t, dhcpc_roles)
 userdom_use_user_terminals(dhcpc_t)
 userdom_dontaudit_search_user_home_dirs(dhcpc_t)
 
+mls_file_read_to_clearance(dhcpc_t)
+
 ifdef(`distro_redhat', `
 	files_exec_etc_files(dhcpc_t)
 ')
-- 
2.17.1

